File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        
        HashMap<Integer,Boolean> numsMap = new HashMap();
        int result = 0;
        
        for(int num : nums){
            
            if(numsMap.get(num) != null && numsMap.get(num)){
                numsMap.put(num,false);
            }else{
                numsMap.put(num,true);
            }
            
        }    

        for(int key : numsMap.keySet()){
            if(numsMap.get(key)){
                result = key;
            }
        }
      
        return result;
        
    }
}

File: 0136-single-number/0136-single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
     if(nums.length === 1) return nums[0];

    var myMap = new Map();

    for(let i=0; i<nums.length; i++){
        myMap.set(nums[i] , ( myMap.get(nums[i]) || 0 ) + 1)
    }

    for(const [key, value] of myMap) {
        if(value === 1) return key;
    }
    
};